FirebaseSessions/Sources/SessionStartEvent.swift

```diff
@@ -0,0 +1,60 @@
+//
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+@_implementationOnly import GoogleDataTransport
+
+///
+/// SessionStartEvent is responsible for:
+///   1) Writing fields to the Session proto
+///   2) Synthesizing itself for persisting to disk and logging to GoogleDataTransport
+///
+class SessionStartEvent: NSObject, GDTCOREventDataObject {
```

Comment:
Design suggestion: I would recommend keeping proto conversion away from the actual event. SessionStartEvent as a class can have all the essentials for the event and a separate layer can be used to convert the SessionStartEvent to a SessionEventProto and can be used for dispatch.

This would help in moving the session start event across the library until we are ready for dispatch. I would look at this event as meta data preparation in this phase and a dispatch is when we need to turn them to  proto.

Reply:
Separation is always hard for such situations. If we can, I would recommend doing the separation now.